pkg/controller/daemon/daemoncontroller.go

```diff
@@ -741,6 +749,9 @@ func (dsc *DaemonSetsController) nodeShouldRunDaemonPod(node *v1.Node, ds *exten
 
 	// TODO: Move it to the predicates
 	for _, c := range node.Status.Conditions {
+		if critical {
+			break
+		}
 		if c.Type == v1.NodeOutOfDisk && c.Status == v1.ConditionTrue {
```

Comment:
not for this PR: There are other too that the DaemonSet should ideally respect - `MemoryPressure`, `DiskPressure` for example are some of them.